File: python/examples/count_enrollments.py
#!/usr/bin/python
# written by andrewt@cse.unsw.edu.au as a 2041 lecture example
# count how many people enrolled in each course

import fileinput, re

course_names = {}
for line in open("course_codes"):
    m = re.match(r'(\S+)\s+(.*\S)', line)
    if m:
        course_names[m.group(1)] = m.group(2);

count = {}
for line in fileinput.input():
    course = re.sub(r'\|.*\n', '', line)
    if course in count:
        count[course] += 1
    else:
        count[course] = 1

for course in sorted(count.keys()):
    print  "%s has %s students enrolled"%(course_names[course], count[course])

File: perl/examples/count_enrollments.pl
#!/usr/bin/perl -w
# written by andrewt@cse.unsw.edu.au as a 2041 lecture example
# count how many people enrolled in each course

open F, "<course_codes" or die "$0: can not open course_codes: $!";
while ($line = <F>) {
    chomp $line;
    $line =~ /([^ ]+) (.+)/ or die "$0: bad line format '$line'";
    $course_names{$1} = $2;
}
close F;

while ($course = <>) {
    chomp $course;
    $course =~ s/\|.*//;
    $count{$course}++;
}

foreach $course (sort keys %count) {
    print "$course_names{$course} has $count{$course} students enrolled\n";
}
